src/helpers/color.ts

```diff
@@ -367,21 +367,132 @@ const ColorsList = [
   "rgb(158,218,229)",
 ];
 
-export function getNthColor(index: number): Color {
-  return ColorsList[index % ColorsList.length];
+const COLORS_SM = [
+  "#4EA7F2", // Blue
+  "#EA6175", // Red
+  "#43C5B1", // Teal
+  "#F4A261", // Orange
+  "#8481DD", // Purple
+  "#FFD86D", // Yellow
+];
+const COLORS_MD = [
+  "#4EA7F2", // Blue #1
+  "#3188E6", // Blue #2
+  "#43C5B1", // Teal #1
+  "#00A78D", // Teal #2
+  "#EA6175", // Red #1
+  "#CE4257", // Red #2
+  "#F4A261", // Orange #1
+  "#F48935", // Orange #2
+  "#8481DD", // Purple #1
+  "#5752D1", // Purple #2
+  "#FFD86D", // Yellow #1
+  "#FFBC2C", // Yellow #2
+];
+const COLORS_LG = [
+  "#4EA7F2", // Blue #1
+  "#3188E6", // Blue #2
+  "#056BD9", // Blue #3
+  "#A76DBC", // Violet #1
+  "#7F4295", // Violet #2
+  "#6D2387", // Violet #3
+  "#EA6175", // Red #1
+  "#CE4257", // Red #2
+  "#982738", // Red #3
+  "#43C5B1", // Teal #1
+  "#00A78D", // Teal #2
+  "#0E8270", // Teal #3
+  "#F4A261", // Orange #1
+  "#F48935", // Orange #2
+  "#BE5D10", // Orange #3
+  "#8481DD", // Purple #1
+  "#5752D1", // Purple #2
+  "#3A3580", // Purple #3
+  "#A4A8B6", // Gray #1
+  "#7E8290", // Gray #2
+  "#545B70", // Gray #3
+  "#FFD86D", // Yellow #1
+  "#FFBC2C", // Yellow #2
+  "#C08A16", // Yellow #3
+];
+const COLORS_XL = [
+  "#4EA7F2", // Blue #1
+  "#3188E6", // Blue #2
+  "#056BD9", // Blue #3
+  "#155193", // Blue #4
+  "#A76DBC", // Violet #1
+  "#7F4295", // Violet #1
```

Comment:
I see you copied odoo/web, but there's probably a smarter way to do that that doesn't involve repeating 4 times Blue #1, 3 times Blue #2, 2 times Blue #3 ...

Reply:
It's complicated to find a way (apart defined the colors somewhere and using the constant), because the order of the colors change with the palette (SM = BRTOPY, MD = BTROPY, LG = XL = BVRTOPGY) so we can't rely on some rule like SM[i] = XL[4*i]. But we can have the XL array defined, and then defined the others according to this one, or having some constant like BLUE_1 and then building the array with these constant. What do you think ?